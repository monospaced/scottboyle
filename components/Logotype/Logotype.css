@import url('../../theme/vars.css');

.Logotype {
  margin: 0;
}

.Logotype-link,
.Logotype-link:visited {
  color: color(var(--background-color) contrast(100%));
}

.Logotype-subtitle {
  display: block;
  color: var(--theme-color);
}
