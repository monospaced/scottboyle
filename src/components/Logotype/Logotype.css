@import url("../../theme/vars.css");

.Logotype {
  margin: 0;
  position: relative;
  top: calc(2rem / 16);
}

.Logotype-link {
  color: var(--text-emphasis-color);
  display: block;
}

.Logotype-link:focus {
  outline: 0;
}

.Logotype-link:hover {
  text-decoration: none;
}

.Logotype-link:active {
  color: var(--text-emphasis-color);
}

.Logotype-subtitle {
  color: var(--theme-color);
  display: block;
}
