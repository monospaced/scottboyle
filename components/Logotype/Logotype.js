import React from 'react';
import {IndexLink} from 'react-router';
import './Logotype.css';

const Logotype = ({title, subtitle}) => {
  return (
    <h1 className="Logotype">
      <IndexLink to="/" className="Logotype-link">
        {title} <strong className="Logotype-subtitle">{subtitle}</strong>
      </IndexLink>
    </h1>
  );
};

export default Logotype;
