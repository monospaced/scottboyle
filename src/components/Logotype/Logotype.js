import { IndexLink } from "react-router";
import React from "react";

import "./Logotype.css";

const Logotype = ({ subtitle, title }) => {
  return (
    <h1 className="Logotype">
      <IndexLink className="Logotype-link" to="/">
        {title} <strong className="Logotype-subtitle">{subtitle}</strong>
      </IndexLink>
    </h1>
  );
};

export default Logotype;
