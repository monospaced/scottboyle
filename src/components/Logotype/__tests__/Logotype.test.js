import React from "react";
import { render } from "enzyme";

import Logotype from "../Logotype.js";

describe("Logotype component", () => {
  it("should render correctly", () => {
    const component = render(<Logotype subtitle="Subtitle" title="Title" />);
    expect(component).toMatchSnapshot();
  });
});
