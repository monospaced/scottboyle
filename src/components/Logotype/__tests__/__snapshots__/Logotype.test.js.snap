// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Logotype component should render correctly 1`] = `
<h1
  class="Logotype"
>
  <a
    class="Logotype-link"
  >
    Title 
    <strong
      class="Logotype-subtitle"
    >
      Subtitle
    </strong>
  </a>
</h1>
`;
